tests/inference_on_device_test.py

```diff
@@ -345,6 +347,9 @@ def test_nograd_after_inference_train(inference_method) -> None:
         show_progress_bars=False,
     )
 
+    prior, _, prior_returns_numpy = process_prior(prior)
+    simulator = process_simulator(simulator, prior, prior_returns_numpy)
+
     theta, x = simulate_for_sbi(simulator, prior, 32)
```

Comment:
as a example to my general comment on the PR: Here, we could just do
```suggestion
    num_simulations = 32
    theta = prior.sample((num_simulations,))
    x = simulator(theta)
```

because we just know that the above prior and simulator are correct (`Distribution`, batched, returns `Tensor` etc.)